llvm/include/llvm/Support/MathExtras.h

```diff
@@ -461,9 +467,11 @@ inline int64_t mod(int64_t Numerator, int64_t Denominator) {
   return Mod < 0 ? Mod + Denominator : Mod;
 }
 
-/// Returns the integer nearest(Numerator / Denominator).
+/// Returns (Numerator / Denominator) rounded by round-half-up. Guranteed to
+/// never overflow.
 inline uint64_t divideNearest(uint64_t Numerator, uint64_t Denominator) {
-  return (Numerator + (Denominator / 2)) / Denominator;
+  return (Numerator / Denominator) +
+         (Denominator == 1 ? 0 : Numerator % Denominator >= Denominator / 2);
```

Comment:
This doesn't look right. It looks like `3 / 7` would round up to `1` with this implementation.

Reply:
Damn, I thought about this for a long time, but didn't get the right expression. Would you kindly help?

Reply:
The issue is that I can't do `2 * (Numerator % Denominator)`, due to possible overflow.

Reply:
Figured it out. I require `& ~(Denominator &1)`.